### PR TITLE
Fix disappearance issue of renamed schema in File metastore

### LINKED ISSUE #13306: Renamed schema disappears in file metastore
Steps to reproduce on `HiveQueryRunner`
```sql
trino> SHOW SCHEMAS IN hive;
       Schema
--------------------
 information_schema
 tpch
 tpch_bucketed
(3 rows)

trino> ALTER SCHEMA hive.tpch RENAME TO test;
RENAME SCHEMA

trino> SHOW SCHEMAS IN hive;
       Schema
--------------------
 information_schema
 tpch_bucketed
(2 rows)
```

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fix disappearance issue of renamed schema in File metastore
Fixes #13306

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Review by findinpath — APPROVED** [review]

**Comment by findinpath** [comment]
Build is red.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java [hunk 1]
```diff
@@ -234,11 +234,16 @@ public synchronized void renameDatabase(String databaseName, String newDatabaseN
...
-        getRequiredDatabase(databaseName);
+        Database database = getRequiredDatabase(databaseName);
         verifyDatabaseNotExists(newDatabaseName);
 
+        Path oldDatabaseMetadataDirectory = getDatabaseMetadataDirectory(databaseName);
+        Path newDatabaseMetadataDirectory = getDatabaseMetadataDirectory(newDatabaseName);
         try {
-            if (!metadataFileSystem.rename(getDatabaseMetadataDirectory(databaseName), getDatabaseMetadataDirectory(newDatabaseName))) {
+            writeSchemaFile(DATABASE, newDatabaseMetadataDirectory, databaseCodec, new DatabaseMetadata(currentVersion, database), false);
+            deleteSchemaFile(DATABASE, oldDatabaseMetadataDirectory);
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java:244]: Why not rename?

**Comment by ebyhr** [comment]
CI hit #12405

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java]: @findepi Changed to rename.

**Review by findepi — APPROVED** [review]

**Pull request merged**